com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationUtilities.java

```diff
@@ -29,13 +29,14 @@
 import static com.unity.androidnotifications.UnityNotificationManager.KEY_NOTIFICATION;
 import static com.unity.androidnotifications.UnityNotificationManager.KEY_REPEAT_INTERVAL;
 import static com.unity.androidnotifications.UnityNotificationManager.KEY_SMALL_ICON;
+import static com.unity.androidnotifications.UnityNotificationManager.KEY_SHOW_IN_FOREGROUND;
 import static com.unity.androidnotifications.UnityNotificationManager.TAG_UNITY;
 
 public class UnityNotificationUtilities {
     // magic stands for "Unity Mobile Notifications Notification"
     private static final byte[] UNITY_MAGIC_NUMBER = new byte[] { 'U', 'M', 'N', 'N'};
     private static final byte[] UNITY_MAGIC_NUMBER_PARCELLED = new byte[] { 'U', 'M', 'N', 'P'};
-    private static final int NOTIFICATION_SERIALIZATION_VERSION = 0;
+    private static final int NOTIFICATION_SERIALIZATION_VERSION = 1;
```

Comment:
Do you have a test which would show a failure if you would forget to bump this number?

Reply:
How do you even write such test? Other than having a expected blob in test.

Reply:
I had a similar case, I was simply checking the if a specific version corresponds to specific size of serialized data.

I assume here, version 1 produces a specific size of data.

Thus you can simply add a check for failure, for cases where you're serializing more data, but forgot to bump the version. In that case the size will be different than the one you expect for version 1.

Reply:
If somebody else picks up this project, and expands serializeNotificationCustom function with extra data, but forgets to update NOTIFICATION_SERIALIZATION_VERSION, then he unknowingly creates a bug. Basically only you know that you need to bump this number, nobody else knows.

None of your current tests will catch such oversight?

Reply:
Added a comment. If you change the serialization without version check at deserialization, the existing v0 test should blow. Add if (NOTIFICATION_SERIALIZATION_VERSION >...) I hope will be itching eyes enough to go look up what a version is and find that comment :)

Reply:
Still think it's quite easy to make an error. Just consider this:
* I add a new variable in AndroidNotification, newVariable
* I add new line for serializing newVariable (in n serializeNotificationCustom ?)
* I add new line for deserializing newVariable

                if (version > 0)
                {
                    showInForeground = in.readBoolean();
                    newVariable= in.readSomething();
                } 
* I forget to bump NOTIFICATION_SERIALIZATION_VERSION, didn't read the comment

And this where it will fail, assuming previous app install serialized notification data. And no test will warn me, that I did the failure.

From what I can say serializeNotificationCustom_v0 only validates if we correctly deserialize data from version 0, but it doesn't validate the future changes.